spark/src/test/scala/org/apache/spark/sql/delta/DeltaParquetFileFormatSuite.scala

```diff
@@ -16,140 +16,214 @@
+
+          val relation = HadoopFsRelation(
+            fileIndex,
+            fileIndex.partitionSchema,
+            readingSchema,
+            bucketSpec = None,
+            deltaParquetFormat,
+            options = Map.empty)(spark)
+          val plan = LogicalRelation(relation)
+
+          if (readIsRowDeletedCol) {
+            val (deletedColumnValue, notDeletedColumnValue) = (1, 0)
+            if (enableDVs) {
+              // Select some rows that are deleted and some rows not deleted
+              // Deleted row `value`: 0, 200, 300, 756, 10352, 19999
+              // Not deleted row `value`: 7, 900
+              checkDatasetUnorderly(
+                Dataset.ofRows(spark, plan)
+                  .filter("value in (0, 7, 200, 300, 756, 900, 10352, 19999)")
+                  .select("value", DeltaParquetFileFormat.IS_ROW_DELETED_COLUMN_NAME)
+                  .as[(Int, Int)],
+                (0, deletedColumnValue),
+                (7, notDeletedColumnValue),
+                (200, deletedColumnValue),
+                (300, deletedColumnValue),
+                (756, deletedColumnValue),
+                (900, notDeletedColumnValue),
+                (10352, deletedColumnValue),
+                (19999, deletedColumnValue))
+            } else {
+              checkDatasetUnorderly(
+                Dataset.ofRows(spark, plan)
+                  .filter("value in (0, 7, 200, 300, 756, 900, 10352, 19999)")
+                  .select("value", DeltaParquetFileFormat.IS_ROW_DELETED_COLUMN_NAME)
+                  .as[(Int, Int)],
+                (0, notDeletedColumnValue),
+                (7, notDeletedColumnValue),
+                (200, notDeletedColumnValue),
+                (300, notDeletedColumnValue),
+                (756, notDeletedColumnValue),
+                (900, notDeletedColumnValue),
+                (10352, notDeletedColumnValue),
+                (19999, notDeletedColumnValue))
+            }
           }
 
-          val df = Dataset.ofRows(spark, plan)
-          assert(rowIndexes(df) === Seq.range(0, 20000).toSet)
+          if (readRowIndexCol) {
+            def rowIndexes(df: DataFrame): Set[Long] = {
+              val colIndex = if (readIsRowDeletedCol) 2 else 1
+              df.collect().map(_.getLong(colIndex)).toSet
+            }
+
+            val df = Dataset.ofRows(spark, plan)
+            assert(rowIndexes(df) === Seq.range(0, 20000).toSet)
 
-          assert(
-            rowIndexes(
-              df.filter("value in (0, 7, 200, 300, 756, 900, 10352, 19999)")) ===
-            Seq(0, 7, 200, 300, 756, 900, 10352, 19999).toSet)
+            assert(
+              rowIndexes(
+                df.filter("value in (0, 7, 200, 300, 756, 900, 10352, 19999)")) ===
+                Seq(0, 7, 200, 300, 756, 900, 10352, 19999).toSet)
+          }
         }
       }
     }
   }
 
+  testWithBothParquetReaders("Existence of Deletion Vectors metadata columns when we " +
```

Comment:
This test probably becomes moot if you don't pass the DV information via the metadata struct anymore 9see my previous comment)